salt/modules/zypper.py

```diff
@@ -115,6 +115,11 @@ def info_installed(*names):
         t_nfo = dict()
         # Translate dpkg-specific keys to a common structure
         for key, value in pkg_nfo.items():
+            try:
+                value = value.encode('UTF-8', errors='replace')
+            except(UnicodeDecodeError):
+                log.error('Package ' + pkg_name + ' has bad UTF-8 code in ' + key + ': ' + value)
```

Comment:
Well, this fails with strings containing invalide utf-8 sequences with:
    TypeError: not all arguments converted during string formatting
    Logged from file zypper.py, line 119
    Traceback (most recent call last):
      File "/usr/lib64/python2.7/logging/__init__.py", line 859, in emit
        msg = self.format(record)
      File "/usr/lib/python2.7/site-packages/salt/log/mixins.py", line 94, in format
      File "/usr/lib64/python2.7/logging/__init__.py", line 732, in format
        return fmt.format(record)
      File "/usr/lib64/python2.7/logging/__init__.py", line 471, in format
        record.message = record.getMessage()
      File "/usr/lib64/python2.7/logging/__init__.py", line 335, in getMessage
        msg = msg % self.args
    TypeError: not all arguments converted during string formatting
    Logged from file zypper.py, line 119

Back to square one ?!